File: src/core/KartMove/External.lua
-- External struct的Lua等效实现
local UnityMath = require(script.Parent.Parent.KartShared.UnityMath)
local Vector3 = UnityMath.Vector3

local External = {}
External.__index = External

function External.new()
    local self = setmetatable({}, External)
    self:Initialize()
    return self
end

function External:Initialize()
    self.slip = false
    self.dragFactor = 1.0
    self.compensationDragFactor = 1.0
    self.wheelFactor = 1.0
    self.annexForce = Vector3.zero
    self.force = Vector3.zero
    self.torque = Vector3.zero
    self.upDownTime = 0.0
    self.upDownLastTime = 0.0
    self.gravityFactor = 1.0
    self.speedLimit = 0.0
    
    -- 这些字段在Initialize中没有设置，但在struct中存在
    self.upDownInterval = 0.0
    self.upDownForce = Vector3.zero
    self.upDownForceIndex = 0
    self.liftVel = Vector3.zero
end

return External

File: sources/paopao/External/External.cs
using System;
using UnityEngine;

public struct External
{
	public void Initialize()
	{
		this.slip = false;
		this.dragFactor = 1f;
		this.compensationDragFactor = 1f;
		this.wheelFactor = 1f;
		this.annexForce = Vector3.zero;
		this.force = Vector3.zero;
		this.torque = Vector3.zero;
		this.upDownTime = 0f;
		this.upDownLastTime = 0f;
		this.gravityFactor = 1f;
		this.speedLimit = 0f;
	}

	public bool slip;

	public float dragFactor;

	public float compensationDragFactor;

	public float wheelFactor;

	public Vector3 annexForce;

	public Vector3 force;

	public Vector3 torque;

	public float gravityFactor;

	public float speedLimit;

	public float upDownTime;

	public float upDownLastTime;

	public float upDownInterval;

	public Vector3 upDownForce;

	public uint upDownForceIndex;

	public Vector3 liftVel;
}
